api/connector/simpleconnector_test.go

```diff
@@ -100,3 +102,29 @@ func (s *simpleConnectorSuite) TestNewSimpleRespectsClientCredentials(c *gc.C) {
 		}
 	}
 }
+
+func (s *simpleConnectorSuite) TestSimpleConnector_Connect(c *gc.C) {
+	connector, err := NewSimple(SimpleConfig{
+		Username:            "alice@canonical.com",
+		ControllerAddresses: []string{"localhost:17080"},
+	})
+	c.Assert(err, gc.IsNil)
+
+	openFunc := func(i *api.Info, do api.DialOpts) (api.Connection, error) {
+		// Zeros to false, ensure it is true after Connect dial opt.
+		c.Assert(do.InsecureSkipVerify, gc.Equals, true)
+
+		// Defaults to 10 * time.Minute, ensure it is overwritten after Connect dial opt.
+		c.Assert(do.Timeout, gc.Equals, 5*time.Minute)
+		return nil, nil
+	}
```

Comment:
We need to assert that this is even called.

```suggestion
	var called bool
	openFunc := func(i *api.Info, do api.DialOpts) (api.Connection, error) {
		called = true
		// Zeros to false, ensure it is true after Connect dial opt.
		c.Assert(do.InsecureSkipVerify, gc.Equals, true)

		// Defaults to 10 * time.Minute, ensure it is overwritten after Connect dial opt.
		c.Assert(do.Timeout, gc.Equals, 5*time.Minute)
		return nil, nil
	}

	... run connect call
	
	c.Assert(called. jc.IsTrue)
```